### PR TITLE
[WIP] Pgsql pod reset

### PR DESCRIPTION
**Pull request by OdedViner**
Signed-off-by: Oded Viner <odedviner@localhost.localdomain>

### CODE REVIEW

File: ocs_ci/utility/workloads/helpers.py [hunk 1]
```diff
@@ -4,11 +4,27 @@
...
+from ocs_ci.ocs.ocp import OCP
+from ocs_ci.utility import templating, utils
+from ocs_ci.utility.spreadsheet.spreadsheet_api import GoogleSpreadSheetAPI
+
 
 log = logging.getLogger(__name__)
 
 DISTROS = {"Debian": "apt-get", "RHEL": "yum"}
 
+Pod_Dic = {
```
Comment (ebenahar) [inline ocs_ci/utility/workloads/helpers.py:21]: Change to POD_DICT

File: ocs_ci/utility/workloads/helpers.py [hunk 2]
```diff
@@ -31,3 +47,88 @@ def find_distro(io_pod):
             log.debug(f"Distro is not {distro}")
         else:
             return distro
+
+
+class PgsqlE2E(object):
```
Comment (ebenahar) [inline ocs_ci/utility/workloads/helpers.py:52]: please add docstrings to the class and the methods

File: tests/e2e/workloads/test_pgsql_reset_pod.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        # Create resource file for pgbench workload and wait for pgbench pod to be created
+        pod_test.create_benchmark()
+        # Delete relevant pod and wait for pgbench_client to complete
+        pod_test.reset_pod()
+        # Running pgbench and parsing logs
+        pod_test.run_pgbench()
+        # Clean up pgbench benchmark
+        pod_test.delete_pgbench()
+        # Collect data and export to Google doc spreadsheet
+        pod_test.collect_log_to_google()
```
Comment (ebenahar) [inline tests/e2e/workloads/test_pgsql_reset_pod.py:89]: really nice and clean!

Comment (hnallurv) [reply tests/e2e/workloads/test_pgsql_reset_pod.py:89]: :-) Yep!

File: ocs_ci/utility/workloads/helpers.py [hunk 4]
```diff
@@ -31,3 +47,88 @@ def find_distro(io_pod):
...
+    def __init__(self, transactions, pod_name, namespace):
+        self.transactions = transactions
```
Comment (vasukulkarni) [inline ocs_ci/utility/workloads/helpers.py:54]: I think we should discuss again, there is a quite bit of code that can be reused  and this class doesn't have to be in helpers.py,

File: ocs_ci/utility/workloads/helpers.py [hunk 5]
```diff
@@ -31,3 +47,88 @@ def find_distro(io_pod):
...
+        pod_obj.wait_for_resource(
+            condition='Completed',
+            resource_name=self.pgbench_client_pod,
+            timeout=timeout,
+            sleep=10,
+        )
+
+    def run_pgbench(self):
+        # Running pgbench and parsing logs
+        output = run_cmd(f'oc logs {self.pgbench_client_pod}')
```
Comment (RazTamir) [inline ocs_ci/utility/workloads/helpers.py:103]: please use ocp.get_logs method

File: ocs_ci/utility/workloads/helpers.py [hunk 6]
```diff
@@ -31,3 +47,88 @@ def find_distro(io_pod):
...
+        for lat in self.pg_output:
+            lat_avg = lat['latency_avg']
+            lat_stddev = lat['lat_stddev']
+            tps_incl = lat['tps_incl']
+            tps_excl = lat['tps_excl']
+            g_sheet.insert_row(
+                [int(lat_avg),
+                 int(lat_stddev),
+                 int(tps_incl),
+                 int(tps_excl)], 2)
```
Comment (RazTamir) [inline ocs_ci/utility/workloads/helpers.py:134]: please fix indentations:
```
g_sheet.insert_row(
    ...

)

```

File: tests/e2e/workloads/test_pgsql_reset_pod.py [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+    # Create RipSaw Operator
+    ripsaw = RipSaw()
+
+    def teardown():
+        ripsaw.cleanup()
+    request.addfinalizer(teardown)
+    return ripsaw
+
+
+@ignore_leftovers
```
Comment (RazTamir) [inline tests/e2e/workloads/test_pgsql_reset_pod.py:26]: any good reason to ignore leftovers?

File: tests/e2e/workloads/test_pgsql_reset_pod.py (hunk 3 above)
Comment (RazTamir) [reply tests/e2e/workloads/test_pgsql_reset_pod.py:89]: +1 - really nice!

**Pull request closed without merging**